plugins/github-actions/src/api/GithubActionsClient.ts

```diff
@@ -24,41 +26,62 @@ import {
 } from '@octokit/types';
 
 export class GithubActionsClient implements GithubActionsApi {
+  private readonly configApi: ConfigApi;
+  private readonly githubAuthApi: OAuthApi;
+
+  constructor(options: { configApi: ConfigApi; githubAuthApi: OAuthApi }) {
+    this.configApi = options.configApi;
+    this.githubAuthApi = options.githubAuthApi;
+  }
+
+  private async getOctokit(hostname?: string): Promise<Octokit> {
```

Comment:
We'd really want this to use the full URL, as we can't guarantee that we'll be able to use only the hostname to select the config forever. We've already seen some cases with GitHub Apps were we might need to select based on owner/repo information in some cases too.

If you're up for it it would be better to be passing URLs around so we can avoid that refactor later. It could just be a hardcoded `https://${hostname}` for now if you don't wanna figure out the `TODO` of grabbing the repo URL

Reply:
1. Could you outline an idea on how to get the repo URL? As far as I've understood there is as of yet no decision on how to annotate entities with github hostname? E.g. the `github.com/project-slug` only gives `owner/repo`?
2. Do you suggest modifying the method signatures from `hostname, owner, repo, ...` to `url, ...`? In the current code that sounds like an extra parse step, but on the other hand it sounds like you are thinking of replacing the `project-slug` annotation to a full URL instead, making it natural to just pass that reference onwards?

Reply:
Yeah getting the actual URL would be awkward. I think `backstage.io/managed-by-location` could work in practice but isn't semantically correct. We would indeed either want to guarantee that that works, or use something like `github.com/repo-url`.

Was actually thinking `url, owner, repo, ...` as it's easier to remove owner and repo later, but tbh let's just stick to what you have as I suspect we won't want to refactor this until around the time we'd have a proper integrations API for use in the frontend anyway. At that point it's prolly a bit of a different change